### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by jamesamuel** [elastic/app-search-javascript#27]
Please add typescript support!

**Comment by antoinewg** [comment]
Any updates on this ? I would love to help if needed

**Comment by oussamarouabah** [comment]
Any progress here!

**Comment by MartinSeeler** [comment]
Bump, please support typescript. It's 2022.

**Comment by quanglam2807** [comment]
Bump! We need this!

**Comment by JasonStoltz** [comment]
Yep, you're absolutely right, we should have TS support, sorry for the delay. We're releasing a new JS client soon that will supersede this one. It will have TS support.

**Comment by JamesLove002** [comment]
When will this new JS client be out?

**Comment by omonk** [comment]
@JasonStoltz - any update when this new library might be released?

**Comment by JasonStoltz** [comment]
Sorry ya'll, it got delayed a bit. The repository is here, it should be out in a pretty stable beta soon. https://github.com/elastic/enterprise-search-js.

**Comment by t2farrell** [comment]
Just created a new issue on that repo.  It appears it hasn't had any activity lately, would be great to see it get some attention.

**Comment by jenseo** [comment]
@JasonStoltz Is the client you link to above (enterprise-search-js) the one we are supposed to use for enterprise app search in node? There are so many different clients, and the one you link has only had one release and is missing things that are available in the documentation (like multisearch for instance).

No one answers the issues and certain endpoint paths (like searchEsSearch for instance) points to /v0/ but in later elastic search endpoints these are upgraded to /v1/. 

Do you have any insights when we can expect a new version in npm of the package you are suggesting above?
